src/config.ts

```diff
@@ -33,20 +43,48 @@ const defaultConfig: Config = {
     artifactsDir: 'build/contracts',
   },
 
-  blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
-  gasPrice: 20e9, // 20 gigawei
-
-  setupProvider: async baseProvider => baseProvider,
+  setupProvider: async (baseProvider) => baseProvider,
 
   coverage: false,
+
+  node: {
+    allowUnlimitedContractSize: false,
+    gasLimit: DEFAULT_BLOCK_GAS_LIMIT,
+    gasPrice: 20e9, // 20 gigawei
+  },
 };
 
 function getConfig(): Config {
   const location = findUp.sync(CONFIG_FILE, { type: 'file' });
   const providedConfig: Partial<Config> = location !== undefined && fs.existsSync(location) ? require(location) : {};
 
+  if (providedConfig.blockGasLimit !== undefined) {
+    warn(`blockGasLimit is deprecated. Use node.gasLimit instead. See ${configHelpUrl} for details.`);
+  }
+
+  if (providedConfig.gasPrice !== undefined) {
+    warn(`Please move gasPrice option inside node option. See ${configHelpUrl} for more details.`);
+  }
+
+  if (!!providedConfig.gasPrice && !!providedConfig.node?.gasPrice) {
+    throw new Error(
+      `GasPrice is specified twice in config. Please fix your config. See ${configHelpUrl} for more details.`,
+    );
+  }
+
+  if (!!providedConfig.blockGasLimit && !!providedConfig.node?.gasLimit) {
+    throw new Error(
+      `GasLimit is specified twice in config. Please fix your config. See ${configHelpUrl} for more details.`,
+    );
+  }
+
   const config: Config = merge(defaultConfig, providedConfig);
 
+  if (config.gasPrice) config.node.gasPrice = config.gasPrice;
```

Comment:
Same issue as above.

```suggestion
  if (config.gasPrice !== undefined) config.node.gasPrice = config.gasPrice;
```